packages/core/streams/anthropic-stream.ts

```diff
@@ -25,9 +50,48 @@ function parseAnthropicStream(): (data: string) => string | void {
   }
 }
 
+/**
+ * Intended for v2023-06-01 and greater,
+ * but may need to be adjusted for future versions.
+ * https://docs.anthropic.com/claude/reference/versioning#version-history
+ */
+function parseAnthropicStream(): (data: string) => string | void {
+  return data => {
+    const json = JSON.parse(data as string) as CompletionChunk
+    const delta = json.completion
+    return delta
+  }
+}
+
+async function* streamable(stream: AsyncIterable<CompletionChunk>) {
+  for await (const chunk of stream) {
+    const text = chunk.completion
+    if (text) yield text
+  }
+}
+
+/**
+ * Accepts either a fetch Response from the Anthropic `POST /v1/complete` endpoint,
+ * or the return value of `await client.completions.create({ stream: true })`
+ * from the `@anthropic-ai/sdk` package.
+ */
 export function AnthropicStream(
-  res: Response,
+  res: Response | AsyncIterable<CompletionChunk>,
   cb?: AIStreamCallbacks
 ): ReadableStream {
-  return AIStream(res, parseAnthropicStream(), cb)
+  if (Symbol.asyncIterator in res) {
+    return readableFromAsyncIterable(streamable(res)).pipeThrough(
+      createCallbacksTransformer(cb)
+    )
+  } else {
+    const apiVersion = res.headers.get('anthropic-version')
```

Comment:
Note this response header does not yet exist, so we shouldn't merge until Anthropic has confirmed they're shipping it.

Reply:
Let us know when to merge here or in slack.

Reply:
I converted this to a draft in the mean time. Thanks @rattrayalex!

Reply:
Fair – checked with Anthropic and it sounds like they won't be able to release this right away, so I added a workaround in the meantime. Should be ready for review/merge now.

Once they add the response header, we can revert 8035f47b605c61dabad578ff204ca4e7e5b22a87